File: dataset/output/GEMINI_gemini-1.5-pro-001/Task55_GEMINI_gemini-1.5-pro-001.cpp
#include <iostream>
#include <regex>
using namespace std;

bool isValidEmail(string email) {
  const regex regexPattern(R"(^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$)");
  return regex_match(email, regexPattern);
}

int main() {
  string email;
  cout << "Enter an email address: ";
  cin >> email;

  if (isValidEmail(email)) {
    cout << "Valid email address" << endl;
  } else {
    cout << "Invalid email address" << endl;
  }

  return 0;
}

File: dataset/output/GEMINI_gemini-1.5-pro-001/Task55_GEMINI_gemini-1.5-pro-001.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class ValidateEmail {
    public static boolean isValidEmail(String email) {
        String regex = "^[A-Za-z0-9+_.-]+@(.+)$";
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(email);
        return matcher.matches();
    }

    public static void main(String[] args) {
        String email = "user@example.com";
        if (isValidEmail(email)) {
            System.out.println("Valid Email");
        } else {
            System.out.println("Invalid Email");
        }
    }
}